### ISSUE TITLE
data presentation bug in staircase mode with points enabled

### ISSUE THREAD

**Issue by anfedorov** [grafana/grafana#5950]
* **I'm submitting a ...**
- [x] Bug report

Please include this information:
- What Grafana version are you using?
v3.1.1
- What datasource are you using?
influx
- What OS are you running grafana on?
linux
- What did you do?
I graphed a bunch of data points
- What was the expected result?
extra points would not be added when I turn on staircase lines
- What happened instead?
extra data points where added when I turned on lines

**IMPORTANT** If it relates to metric data viz:
- An image or text representation of your metric query

<img width="72" alt="good" src="https://cloud.githubusercontent.com/assets/9820/18179740/baa2cbd2-7038-11e6-8ca3-8366f0f61fa5.png"> good

<img width="103" alt="decent" src="https://cloud.githubusercontent.com/assets/9820/18179748/c37ea0a0-7038-11e6-8b94-98f30949ac55.png"> passable

<img width="92" alt="bad" src="https://cloud.githubusercontent.com/assets/9820/18179851/2d3c8e26-7039-11e6-8ef2-720b5722fbc7.png"> NO NO NO NO NOOO

in summary: don't add points to my set of points just because I turn on lines, please! :-)

**Comment by torkelo** [comment]
staircase mode adds points to create the staircase :)

**Comment by torkelo** [comment]
> comment: don't add points to my set of points just because I turn on lines, please! :-)

You did not just turn on lines, you enabled staircase mode :)

**Comment by anfedorov** [comment]
> staircase mode adds points to create the staircase :)

yes, and that is, indeed, the bug.

> You did not just turn on lines, you enabled staircase mode :)

false. staircase mode was already on. staircase mode is the mode of a line, i.e. it is not available when the line is off. quite literally the property is not in the UI when there is no line. staircase *mode* defines the mode of the line, not of the dataset.

**Comment by torkelo** [comment]
Sorry you feel that way. 

Staricase mode is line drawing change. It adds points to create the staircase. The graphing lib cannot render lines without points between them. As you also enabled drawing points you see added staircase points. 

Can you remind me why this is a bug? It feels more like a side effect of the staircase visualization that impacts the points rendering. You will have to choose between them, or duplicate your series, so you have two series, one where you render points and one where you render the staircase line (each series can have individual settings via the series override feature)

**Comment by anfedorov** [comment]
@torkelo
> Can you remind me why this is a bug?

There is absolutely no doubt in my mind about this being a UX bug, although I'm not sure how to explain it because it's so trivially and intuitive correct and I'm not an expert in UX lingo. Most simply, the points of a graph ought to represent data points from that data set. The line connecting points represent an estimation of meaningful values between the dataset. The line could be a rough estimate of a variety of best-fits to the data points. This is how it works every single time you graph points and a line through or around the points, from middle school experiments graphed in Excel to visualizations in peer reviewed scientific journals.

**Comment by anfedorov** [comment]
@torkelo if you have any UX folks on the team, please escalate this question to them

**Comment by torkelo** [comment]
Ok, I will, escalating to @torkelo

**Comment by anfedorov** [comment]
thanks! hopefully he'll get on it asap.
